turbinia/processors/google_cloud.py

```diff
@@ -0,0 +1,412 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Evidence processor for Google Cloud resources."""
+
+import logging
+import os
+import stat
+import time
+import urllib2
+
+from apiclient.discovery import build
+from oauth2client.client import GoogleCredentials
+
+from turbinia import config
+from turbinia import TurbiniaException
+
+log = logging.getLogger('turbinia')
+
+RETRY_MAX = 10
+
+
+class GoogleCloudProject(object):
+  """Class representing a Google Cloud Project.
+
+  Attributes:
+    project_id: Project name.
+    default_zone: Default zone to create new resources in.
+  """
+
+  COMPUTE_ENGINE_API_VERSION = 'v1'
+
+  def __init__(self, project_id, default_zone=None):
+    """Initialize the GoogleCloudProject object.
+
+    Args:
+      project_id: The name of the project.
+      default_zone: Default zone to create new resources in.
+    """
+    self.project_id = project_id
+    self.default_zone = default_zone
+
+  def _CreateService(self, service_name, api_version):
+    """Creates an GCP API service.
+
+    Args:
+      service_name: Name of the service.
+      api_version: Which version of the API to use.
+
+    Returns:
+      API service resource (apiclient.discovery.Resource)
+    """
+    credentials = GoogleCredentials.get_application_default()
+    return build(service_name, api_version, credentials=credentials)
+
+  def _ExecuteOperation(self, service, operation, zone, block):
+    """Executes API calls.
+
+    Args:
+      service: API service respource (apiclient.discovery.Resource).
+      operation: API operation to be executed.
+      zone: GCP zone to execute the operation in. None means GlobalZone.
+      block: Boolean indicating if the opearation shuld block before return.
+
+    Returns:
+      Operation result in JSON format.
+
+    Raises:
+      TurbiniaException: If API call failed.
+    """
+    if not block:
+      return operation
+
+    while True:
+      if zone:
+        result = service.zoneOperations().get(
+            project=self.project_id, zone=zone,
+            operation=operation[u'name']).execute()
+      else:
+        result = service.globalOperations().get(
+            project=self.project_id, operation=operation[u'name']).execute()
+
+      if u'error' in result:
+        raise TurbiniaException(result[u'error'])
+
+      if not block or result[u'status'] == u'DONE':
```

Comment:
Not sure what you mean in this case.  This is something set by the API call, and is just compared here.  Let me know if there's something specific you wanted me to change.